build.sc

```diff
@@ -0,0 +1,88 @@
+import $ivy.`com.goyeau::mill-scalafix:0.2.5`
+import com.goyeau.mill.scalafix.ScalafixModule
+import mill._, scalalib._, scalafmt._
+
+val spinalVersion = "1.6.0"
+
+object poseidon extends ScalaModule with ScalafmtModule with ScalafixModule {
+  def scalaVersion = "2.12.14"
+
+    override def scalacOptions = Seq(
+    //"-V", // Print a synopsis of verbose options.
+    //"-W", // Print a synopsis of warning options.
+    //"-X", // Print a synopsis of advanced options.
+    //"-Y", // Print a synopsis of private options.
+
+    "-deprecation",                      // Emit warning and location for usages of deprecated APIs.
+    "-encoding", "utf-8",                // Specify character encoding used by source files.
+    "-explaintypes",                     // Explain type errors in more detail.
+    "-feature",                          // Emit warning and location for usages of features that should be imported explicitly.
+    "-language:existentials",            // Existential types (besides wildcard types) can be written and inferred
+    "-language:reflectiveCalls",
+    //"-language:experimental.macros",     // Allow macro definition (besides implementation and application)
+    //"-language:higherKinds",             // Allow higher-kinded types
+    //"-language:implicitConversions",     // Allow definition of implicit functions called views
+    "-unchecked",                        // Enable additional warnings where generated code depends on assumptions.
+    "-Xcheckinit",                       // Wrap field accessors to throw an exception on uninitialized access.
+    "-Xfatal-warnings",                  // Fail the compilation if there are any warnings.
+    //"-Xdev",                             // Indicates user is a developer - issue warnings about anything which seems amiss.
+    "-Xverify",                         // Verify generic signatures in generated bytecode.
+    //"-Xplugin-list",                    // Print a synopsis of loaded plugins. [false]
+
+
+    "-Wconf:any:warning-verbose",     // Configure reporting of compiler warnings; use `help` for details.
+    //"-Wdead-code",                    // Warn when dead code is identified. [false]
```

Comment:
do not disable lints, these lints are very important

Reply:
scala version has been changed to 2.13.6 and these lints are enabled